middleware.go

```diff
@@ -44,6 +45,15 @@ func parseRequestURL(c *Client, r *Request) error {
 		}
 	}
 
+	path := reqURL.Path
+	for k, v := range r.Param {
```

Comment:
Let's call this `:id` as `PathParam` and `args` as for those values.

* It's easily error prone for users, so validation is required/improtant
     * PathParam count and args count (refer codesnippet: `countPathParams`) should match otherwise return error
     * Keep sane number, let's limit path param count to `100`, if path params is more than 100 return error
     * if path param is missing value then return error
* Regex is not needed here. Also it's expensive operation
     * refer codesnippet: `composePathURL`

**Please note:** I have not tested these code snippet, but it expected to work. if needs correction please feel free to do so.

**Code Snippet:** `countPathParams`
```go
func countPathParams(path string) uint8 {
  var n uint
  for i := 0; i < len(path); i++ {
  	if path[i] != ':' {
  		continue
  	}
  	n++
  }
  return uint8(n)
}
```

**Code Snippet:** `composePathURL`
```go
// compose path URL with values
pathURL := "/"
for _, segment := range strings.Split(reqURL.Path, "/")[1:] {
  if len(strings.TrimSpace(segment)) == 0 {
    continue
  }

  if segment[0] == ':' {
    argName := segment[1:]
    if arg, found := r.PathParam[argName]; found {
      pathURL = path.Join(pathURL, fmt.Sprintf("%v", arg))
      delete(r.PathParam, argName)
      continue
    }

    return fmt.Errorf("'%v' param not found", segment[1:])
  }

  pathURL = path.Join(pathURL, segment)
}

reqURL.Path = pathURL
```